src/Microsoft.AspNet.Mvc.Core/Rendering/JsonHelper.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.IO;
+using Microsoft.AspNet.Mvc.Core;
+using Newtonsoft.Json;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc.Rendering
+{
+    /// <summary>
+    /// Default implementation of <see cref="IJsonHelper"/>.
+    /// </summary>
+    public class JsonHelper : IJsonHelper, ICanHasViewContext
+    {
+        private ViewContext _viewContext;
+        private JsonOutputFormatter _jsonOutputFormatter;
+
+        /// <inheritdoc />
+        public ViewContext ViewContext
```

Comment:
Sorry I missed fact this property is used only to get the `IServiceProvider`.  Simpler approach is to add a `public JsonHelper(JsonOutputFormatter jsonOutputFormatter)` constructor.  DI will ensure the instance is never created without the formatter.

Then you won't need the `ViewContext` property or the new resource.  May also slightly simplify the test code.